src/directives/formly-field.js

```diff
@@ -77,9 +77,41 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
         return undefined
       }
       if (angular.isDefined(newVal)) {
-        $scope.model[$scope.options.key] = newVal
+        parseSet($scope.options.key, $scope.model, newVal)
+      }
+      return parseGet($scope.options.key, $scope.model)
+    }
+
+    function parseSet(key, model, newVal) {
+      // If either of these are null/undefined then just return undefined
+      if (!key || !model) {
+        return
+      }
+      // If we are working with a number then $parse wont work, default back to the old way for now
+      if (angular.isNumber(key)) {
+        // TODO: Fix this so we can get several levels instead of just one with properties that are numeric
+        model[key] = newVal
+      } else {
+        const setter = $parse($scope.options.key).assign
+        if (setter) {
```

Comment:
I would actually leave it.  Because if you get into a situation where the key you sent in is non numeric but doesn't satisfy a property constraint you will blow up if you try to use a setter where one isn't present.  It wont "work" but it also wont fail either.  Your call.